quantized_mesh_encoder/encode.py

```diff
@@ -70,7 +77,18 @@ def encode(
         ellipsoid,
         Ellipsoid), ('ellipsoid must be an instance of the Ellipsoid class')
 
-    header = compute_header(positions, sphere_method, ellipsoid=ellipsoid)
+    # TODO: Also check no duplicate instances of an extension type
+    assert all(
+        isinstance(ext, ExtensionBase) for ext in extensions
+    ), 'extensions must be list of instances of the Extension class'
+
+    cartesian_positions = to_ecef(positions, ellipsoid=ellipsoid)
+
+    header = compute_header(
+        positions,
+        np.copy(cartesian_positions),
+        sphere_method,
+        ellipsoid=ellipsoid)
```

Comment:
I'm not sure I handled the following in the cleanest way...

Basically I need `cartesian_positions` in both `compute_header()` and `write_vertex_normals()`. So I hoisted the calculation of `cartesian_positions` up into the `encode()` function.

However, I encountered two issues:
- `to_ecef()` was mutating the original `positions` array passed in, so I changed the operations to create new arrays
- somewhere in `computer_header()` the `cartesian_positions` array passed in was also getting mutated so I now pass it a copy.

Reply:
If we use an `encode` method on each extension, then we don't need to make any changes in `encode` with positions. But also I think any method that's mutating the input array should probably be making a copy...